### ISSUE TITLE
Alignment?

### ISSUE THREAD

**Issue by saiz06** [vgteam/vg#1782]
I have a very basic question. Why do we need an alignment to map the vcf to a reference genome? When the vcf is generated, it is already aligned against a reference, so why align it again?

**Comment by adamnovak** [comment]
I don't think you do need an alignment. If you want to use a vcf to make a graph, you can do something like `vg construct -r ref.fa -v ref.vcf > ref.vg` and use the VCF and its corresponding FASTA together, without any aligning.

We do have another command, `vg add`, which adds variants from a VCF into an existing graph. This needs to do some alignment in order to work out how to describe new variants in light of the existing variants. For example, if the graph already contains an insert of `GAT` at a particular point, and the variant you are adding is an insert of `GCT`, the `vg add` tool will do alignment to work out that the best way to represent the new insertion is actually as an A->C SNP on top of the existing insertion. However, most people probably want to be using `vg construct` to build graphs, in general.

**Comment by ekg** [comment]
The VCF and reference represent a DAG implicitly. `vg construct` generates a variation graph from this sequence DAG, and includes the reference sequence in the graph as a path. That path through the graph can equivalently be represented as an alignment to the vg graph we make (in vg's GAM alignment format), but in practice we usually work with it via an index (the .xg index). In other words the alignment records the path we would walk through the graph to reconstruct that particular sequence.

In vg we are usually aligning new sequences into the graphs made from VCFs or the outputs of assembly algorithms like minia3, SGA, canu, or fermikit. But it is also natural to represent paths embedded in the graph as alignments.

Does this help to resolve your question?

**Comment by saiz06** [comment]
If I were to construct a VG graph using the same reference which was used to generate the VCF and then made another VG graph using a different reference, what would be the difference? Will the positions be matched to the References using indexing or alignment?

Also, how does the json incorporate the rest of the information from the VCF?

**Comment by saiz06** [comment]
Thank you, this has been helpful. 

By JSON I meant, the same as you have mentioned, that is if I were to export the graph in a JSON format.

So, if you are not using any other information except the variants, is indexing being done to match a path to a particular VCF file? Or for something else? I was reading that the GCSA2 indexing allows to carry out string searches, so I am assuming that if someone were to be looking for a particular set of sequences, this could be done through searching the index. @ekg mentioned, .xg index may be used for alignment. Would you be able to point me to a reading source so I may better understand this please?

**Comment by glennhickey** [comment]
https://www.biorxiv.org/content/early/2017/12/15/234856

**Comment by faithokamoto** [comment]
Seems you got your questions answered.
